docs/examples/shading/plot_martinez_shade_loss.py

```diff
@@ -0,0 +1,271 @@
+#    5-6.
+#    *Source: César Domínguez. CC BY-SA 4.0, Wikimedia Commons*
+#
+# In the image above, each orange U-shaped string section is a block.
+# By symmetry, the yellow inverted-U's of the subcircuit are also blocks.
+# For this reason, the half-cut cell modules have 6 blocks in total: two along
+# the longest side and three along the shortest side.
+
+blocks_per_module = {
+    "1 bypass diode": 1,
+    "3 bypass diodes": 3,
+    "3 bypass diodes half-cut, portrait": 2,
+    "3 bypass diodes half-cut, landscape": 3,
+}
+
+# Calculate the number of shaded blocks during the day
+shaded_blocks_per_module = {
+    k: np.ceil(blocks_N * shaded_fraction)
+    for k, blocks_N in blocks_per_module.items()
+}
+
+# %%
+# Results
+# -------
+# Now that we have the number of shaded blocks for each module configuration,
+# we can apply the model and estimate the power loss due to shading.
+#
+# Note that this model is not linear with the shaded blocks ratio, so there is
+# a difference between applying it to just a module or a whole row.
+
+shade_losses_per_module = {
+    k: pvlib.shading.direct_martinez(
+        poa_global=poa_global,
+        poa_direct=poa_direct,
+        shaded_fraction=shaded_fraction,
+        shaded_blocks=module_shaded_blocks,
+        total_blocks=blocks_per_module[k],
+    )
+    for k, module_shaded_blocks in shaded_blocks_per_module.items()
+}
+
+shade_losses_per_row = {
+    k: pvlib.shading.direct_martinez(
+        poa_global=poa_global,
+        poa_direct=poa_direct,
+        shaded_fraction=shaded_fraction,
+        shaded_blocks=module_shaded_blocks * N_modules_per_row,
+        total_blocks=blocks_per_module[k] * N_modules_per_row,
+    )
+    for k, module_shaded_blocks in shaded_blocks_per_module.items()
+}
+
+# %%
+# Plotting the results
+# ^^^^^^^^^^^^^^^^^^^^
+
+fig, (ax1, ax2) = plt.subplots(2, 1, sharex=True)
+fig.suptitle("Martinez power losses due to shading")
+for k, shade_losses in shade_losses_per_module.items():
+    linestyle = "--" if k == "3 bypass diodes half-cut, landscape" else "-"
+    ax1.plot(times, shade_losses, label=k, linestyle=linestyle)
+ax1.legend(loc="upper center")
+ax1.grid()
+ax1.set_xlabel("Time")
+ax1.xaxis.set_major_formatter(
+    ConciseDateFormatter("%H:%M", tz="Europe/Madrid")
+)
+ax1.set_ylabel(r"$P_{out}$ losses")
+ax1.set_title("Per module")
+
+for k, shade_losses in shade_losses_per_row.items():
+    linestyle = "--" if k == "3 bypass diodes half-cut, landscape" else "-"
+    ax2.plot(times, shade_losses, label=k, linestyle=linestyle)
+ax2.legend(loc="upper center")
+ax2.grid()
+ax2.set_xlabel("Time")
+ax2.xaxis.set_major_formatter(
+    ConciseDateFormatter("%H:%M", tz="Europe/Madrid")
+)
+ax2.set_ylabel(r"$P_{out}$ losses")
+ax2.set_title("Per row")
+fig.tight_layout()
+fig.show()
+
+# %%
+# Note how the half-cut cell module in portrait behaves worse that the
+# normal module with three bypass diodes. This is because the number of shaded
```

Comment:
```suggestion
# Note how the half-cut cell module in portrait performs better than the
# normal module with three bypass diodes. This is because the number of shaded
```